[RPD-183] Refactor `matcha analytics` command to display help when no argument passed

swells2020 opened the pull request:
This pull request adds default functionality to the `analytics` command, such that the `--help` option will be displayed by default where no other options, arguments, or sub-commands are passed. This is implemented by using typer's built-in `no_args_is_help=True` argument when creating the analytics sub-command branch.

One additional test has been included to test this functionality.

This PR also corrects for missing type definitions in relevant functions and modules.

## Checklist

Please ensure you have done the following:

* [X] I have read the [CONTRIBUTING](https://github.com/fuzzylabs/matcha/blob/main/CONTRIBUTING.md) guide.
* [ ] I have updated the documentation if required.
* [X] I have added tests which cover my changes.

## Type of change

Tick all those that apply:

* [ ] Bug Fix (non-breaking change, fixing an issue)
* [ ] New feature (non-breaking change to add functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [X] Other (add details above)

dudeperf3ct commented on src/matcha_ml/cli/analytics.py:
```diff
@@ -5,7 +5,7 @@
 
 # create a typer app to group all analytics subcommands
-app = typer.Typer()
+app = typer.Typer(no_args_is_help=True)
```
For consistency with main app in `cli.py`, we can disable `pretty_exceptions_show_locals` as well inside the Typer.

swells2020 replied:
Added, thanks! ➕

JonoCX commented on tests/test_cli/test_analytics.py:
```diff
@@ -110,3 +110,18 @@ def test_opt_in_subcommand(
         assert dict(yaml.safe_load(f)) == expected_configuration
+
+
+def test_cli__analytics_command_no_args(runner):
```
There's a double underscore in the function name and missing typing for the `runner`, it should be `runner: TemplateRunner`

JonoCX commented on tests/test_cli/test_analytics.py:
```diff
@@ -110,3 +110,18 @@ def test_opt_in_subcommand(
+
+
+def test_cli__analytics_command_no_args(runner):
+    """Test cli analytics command when no argument passed."""
```
Docstring should include `Args` and the summary for the function could be more description: "Test CLI analytics command displays the help message when no arguments are passed."

JonoCX reviewed (changes requested):
Nice work. 

The title for the PR should match the ticket exactly and the description _could_ include that the functionality added here displays the help message to the user rather than "missing command" (or similar).

See other comments r.e. function name and docstrings

JonoCX reviewed (approved):
LGTM 🚀

swells2020 commented:
Jon, Shubham,

I've added the missing flag like Shubham suggested, thanks.

I've also updated the PR title and description Jon, I hope that's more in line with but do let me know if you'd like additional changes.

I've also added missing types through the `test_analytics` module, and in `conftest`, as well as updating the docstrings and correcting the erroneous function name.

Thank you both for your 👀 !

dudeperf3ct reviewed (approved):
LGTM. Well done @swells2020
